pallas-primitives: Unusable with `default-features = false`

pallas-primitives = {version = "0.35.0", default-features = false }

```
error[E0432]: unresolved import `serde`
  --> /home/jens/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/pallas-primitives-0.35.0/src/plutus_data.rs:11:5
   |
11 | use serde::{Deserialize, Serialize};
   |     ^^^^^ use of unresolved module or unlinked crate `serde`
   |
   = help: if you wanted to use a crate named `serde`, use `cargo add serde` to add it to your `Cargo.toml`

error[E0432]: unresolved import `serde`
 --> /home/jens/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/pallas-primitives-0.35.0/src/alonzo/model.rs:5:5
  |
5 | use serde::{Deserialize, Serialize};
  |     ^^^^^ use of unresolved module or unlinked crate `serde`
  |
  = help: if you wanted to use a crate named `serde`, use `cargo add serde` to add it to your `Cargo.toml`

error[E0432]: unresolved import `serde`
 --> /home/jens/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/pallas-primitives-0.35.0/src/babbage/model.rs:5:5
  |
5 | use serde::{Deserialize, Serialize};
  |     ^^^^^ use of unresolved module or unlinked crate `serde`
  |
  = help: if you wanted to use a crate named `serde`, use `cargo add serde` to add it to your `Cargo.toml`

error[E0432]: unresolved import `serde`
 --> /home/jens/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/pallas-primitives-0.35.0/src/conway/model.rs:5:5
  |
5 | use serde::{Deserialize, Serialize};
  |     ^^^^^ use of unresolved module or unlinked crate `serde`
  |
  = help: if you wanted to use a crate named `serde`, use `cargo add serde` to add it to your `Cargo.toml`

error[E0432]: unresolved import `serde`
  --> /home/jens/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/pallas-primitives-0.35.0/src/lib.rs:21:5
   |
21 | use serde::{Deserialize, Serialize};
   |     ^^^^^ use of unresolved module or unlinked crate `serde`
   |
   = help: if you wanted to use a crate named `serde`, use `cargo add serde` to add it to your `Cargo.toml`

error: cannot find attribute `serde` in this scope
   --> /home/jens/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/pallas-primitives-0.35.0/src/plutus_data.rs:376:3
    |
376 | #[serde(into = "String")]
    |   ^^^^^

error: cannot find attribute `serde` in this scope
   --> /home/jens/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/pallas-primitives-0.35.0/src/plutus_data.rs:377:3
    |
377 | #[serde(try_from = "String")]
    |   ^^^^^
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -92,6 +92,20 @@ jobs:
       - uses: Swatinem/rust-cache@v2
       - run: cargo check --workspace --all-targets ${{ matrix.flags }}
 
+  isolated-features:
+    name: Isolated no-default-features
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: true
+      - uses: dtolnay/rust-toolchain@stable
+      - uses: Swatinem/rust-cache@v2
+      # A workspace-wide --no-default-features build unifies features across
+      # members, so it always pulls pallas-primitives in with `json` enabled.
+      # Check the crate in isolation to catch missing serde feature gating.
+      - run: cargo check -p pallas-primitives --no-default-features
+
   msrv:
     name: MSRV (1.88)
     runs-on: ubuntu-latest

--- a/pallas-primitives/Cargo.toml
+++ b/pallas-primitives/Cargo.toml
@@ -15,7 +15,7 @@ authors.workspace = true
 hex.workspace = true
 pallas-crypto.workspace = true
 pallas-codec.workspace = true
-serde = { workspace = true, optional = true }
+serde = { workspace = true }
 serde_json = { workspace = true, optional = true }
 
 [dev-dependencies]
@@ -23,6 +23,6 @@ proptest = { workspace = true, features = ["alloc"] }
 test-case = "3.3.1"
 
 [features]
-json = ["serde", "serde_json"]
+json = ["serde_json"]
 default = ["json"]
 relaxed = ["pallas-crypto/relaxed"]

